tools/keepalived/keepalived/libipvs-2.6/libipvs.c

```diff
@@ -611,55 +664,63 @@ struct dp_vs_blklst_conf_array *ipvs_get_blklsts(void)
 struct ip_vs_get_dests *ipvs_get_dests(ipvs_service_entry_t *svc)
 {
 	struct ip_vs_get_dests *d;
-	struct ip_vs_get_dests_kern *dk, *dk_rcv;
+	struct dp_vs_get_dests *dpvs_dests, *dpvs_dests_rcv;
+	struct ip_vs_dest_entry *ipvs_entry;
+	struct dp_vs_dest_entry *dpvs_entry;
 	size_t len, len_rcv;
 	int i;
 
-	len = sizeof(*d) + sizeof(ipvs_dest_entry_t) * svc->num_dests;
-	if (!(d = malloc(len)))
+	len = sizeof(struct ip_vs_get_dests) + 
+		sizeof(ipvs_dest_entry_t) * svc->num_dests;
+	if (!(d = calloc(len, 1)))
 		return NULL;
 
-	ipvs_func = ipvs_get_dests;
-
-	if (svc->af != AF_INET) {
-	  errno = EAFNOSUPPORT;
-	  free(d);
-	  return NULL;
-	}
-
-	len = sizeof(*dk) + sizeof(struct ip_vs_dest_entry_kern) * svc->num_dests;
-	if (!(dk = malloc(len))) {
+	len = sizeof(struct dp_vs_get_dests) +
+		sizeof(struct dp_vs_dest_entry) * svc->num_dests;
+	if (!(dpvs_dests = calloc(len, 1))) {
```

Comment:
dpvs_dests is used as sockopt input. I think it needs not to include dest entry length.
`len = sizeof(struct dp_vs_get_dests) `

Reply:
OK